tests/tests/functional/chart_test.py

```diff
@@ -0,0 +1,196 @@
+            print(f'Error from docker loading image: {verifier_image_name}:{image_tag}')
+            return f"FAIL pulling image : docker.errors.APIError: {exc.args}"
+    else:
+        try:
+            verifier_image = client.images.get(f"{verifier_image_name}:{image_tag}")
+        except docker.errors.ImageNotFound:
+            return f"FAIL image not found:  : {verifier_image_name}:{image_tag}"
+        except docker.errors.APIError as exc:
+            return f"FAIL getting image : {verifier_image_name}:{image_tag} : docker.errors.APIError: {exc.args}"
+
+    os.environ["VERIFIER_IMAGE"] = f"{verifier_image_name}:{image_tag}"
+
+    docker_command = "verify "
+    local_chart = False
+    if chart_location.startswith('http:/') or chart_location.startswith('https:/'):
+        docker_command = docker_command + chart_location
+    else:
+        if os.path.exists(chart_location):
+            docker_command = docker_command + f"/charts/{os.path.basename(chart_location)}"
+            local_chart = True
+        else:
+            return f"FAIL: chart does not exist: {os.path.abspath(chart_location)}"
+
+    if profile_type:
+        docker_command = docker_command + " --set profile.vendorType=" + profile_type
+
+    print(f'docker command: {docker_command}')
+
+    client = docker.from_env()
+
+    kubeconfig = os.environ.get("KUBECONFIG")
+    if not kubeconfig:
+        return "FAIL: missing KUBECONFIG environment variable"
+
+    docker_volumes = {kubeconfig: {'bind': '/kubeconfig', 'mode': 'ro'}}
+    docker_environment = {"KUBECONFIG": '/kubeconfig'}
+
+    try:
+        if local_chart:
+            chart_directory = os.path.dirname(os.path.abspath(chart_location))
+            docker_volumes[chart_directory] = {'bind': '/charts/', 'mode': 'rw'}
+
+        output = client.containers.run(verifier_image,docker_command,stdin_open=True,tty=True,stderr=True,volumes=docker_volumes,environment=docker_environment)
+
+    except docker.errors.ContainerError as exc:
+        return f"FAIL: docker.errors.ContainerError: {exc.args}"
+    except docker.errors.ImageNotFound as exc:
+        return f"FAIL: docker.errors.ImageNotFound: {exc.args}"
+    except docker.errors.APIError as exc:
+        return f"FAIL: docker.errors.APIError: {exc.args}"
+
+    if not output:
+        return f"FAIL: no report produced : {docker_command}"
+
+    return output.decode("utf-8")
+
+@then("I should see the report-info from the generated report matching the expected report-info")
+def check_report(run_verifier, profile_type, report_info_location):
+
+    if run_verifier.startswith("FAIL"):
+        pytest.fail(f'FAIL some tests failed: {run_verifier}')
+
+    report_data = yaml.load(run_verifier, Loader=Loader)
+
+    report_vendor_type = report_data["metadata"]["tool"]["profile"]["VendorType"]
+    if report_vendor_type != profile_type:
+        print(f"FAIL: profiles do not match. Expected {profile_type}, but report has {report_vendor_type}")
```

Comment:
Do we still want to continue if this step fails? I mean pytest.fail not needed here?

Reply:
good spot, we do want to continue to see other errors but set a boolean so test would fail even if this was the only failure.